packages/aurora-ether/src/natural-ether/sendToAurora/index.ts

```diff
@@ -177,10 +177,10 @@ export async function findAllTransactions (
     options.etherCustodianAbi ?? bridgeParams.etherCustodianAbi,
     provider
   )
-  const filter = ethTokenLocker.filters.Deposited!(sender)
+  const filter = ethTokenLocker.filters.Deposited!()
   const events = await ethTokenLocker.queryFilter(filter, fromBlock, toBlock)
   const auroraAddress = options.auroraEvmAccount ?? bridgeParams.auroraEvmAccount as string + ':'
-  return events.filter(event => event.args!.recipient.startsWith(auroraAddress)).map(event => event.transactionHash)
+  return events.filter(async event => (await event.getTransaction()).from === sender).filter(event => event.args!.recipient.startsWith(auroraAddress)).map(event => event.transactionHash)
```

Comment:
How many events can there be here and how many requests we may have to do to find the right one?

Reply:
Above, we are extracting only Deposited events for the required contract. After that, it depends on how many transfers we have. For the testnet, there are definitely few enough events; for the mainnet, I'm not sure. It's probably best to check with Pasha.

Reply:
I believe `getTransaction` sends an RPC request to the node, so if we have hundreds of events that will be too much. 

Perhaps instead we should emit events from the Proxy contracts with proper sender and query those?

Reply:
cc @karim-en

Reply:
Yeah, this is a problem; we need to index the events using the real sender.
Let's also add events to the proxy contract and filter events from both of the contracts by the sender.

Reply:
fixed: https://github.com/aurora-is-near/rainbow-bridge-client/pull/97/commits/85fdefd1fce1ac6aa050e766e23f5d700ddf0ea9

Reply:
What about reading events from both Proxy and Implementation as @karim-en suggested?

Reply:
fixed: https://github.com/aurora-is-near/rainbow-bridge-client/pull/97/commits/2fa34ff91e01c38d84f398331ba279e18db25245